check-bundle-size CI fails on fresh runners due to pre-existing TS2769 error in node renderer prepare script

## Problem

The `check-bundle-size` CI workflow fails on fresh runners because `pnpm install` triggers the `prepare` script in `react-on-rails-pro-node-renderer`, which runs `tsc` when `lib/ReactOnRailsProNodeRenderer.js` doesn't exist (no cache). The compilation fails with a pre-existing TS2769 type error.

## Error

```
src/worker.ts(170,12): error TS2769: No overload matches this call.
```

The error is a type incompatibility between `@fastify/multipart`'s `onFile` handler and the HTTP2 server types. The `onFile` callback binds `this: FastifyRequest` (which uses `Http2ServerRequest`), but `@fastify/multipart`'s type definitions only have overloads for HTTP1 (`IncomingMessage`).

## Why it's intermittent

The `prepare` script has a guard:
```json
"prepare": "[ -f lib/ReactOnRailsProNodeRenderer.js ] || (rm -rf ./lib && tsc --project src/tsconfig.json)"
```

- **Fresh runner** (no cached `lib/`): `tsc` runs → fails → `pnpm install` fails → job fails
- **Runner with cached `lib/`**: compilation is skipped → job proceeds normally

## Evidence

- PR #2464 (`check-bundle-size` → failure): fresh runner, no cache
- PR #2456 head commit (`check-bundle-size` → failure): same issue
- PR #2444 (`check-bundle-size` → success): likely had cached `lib/`

## Relevant files

- `.github/workflows/bundle-size.yml` — workflow definition
- `packages/react-on-rails-pro-node-renderer/package.json` (line 89) — `prepare` script
- `packages/react-on-rails-pro-node-renderer/src/worker.ts` (line 170) — type error location
- `packages/react-on-rails-pro-node-renderer/src/worker/types.ts` — `FastifyRequest` type using `Http2Server`

## Possible fixes

1. Fix the type error (add proper HTTP2 overloads or cast)
2. Add `lib/` caching to the `bundle-size.yml` workflow
3. Exclude `react-on-rails-pro-node-renderer` from the bundle size workflow's install step (it only measures the open-source package)

--- a/packages/react-on-rails-pro-node-renderer/src/worker.ts
+++ b/packages/react-on-rails-pro-node-renderer/src/worker.ts
@@ -176,7 +176,9 @@ export default function run(config: Partial<Config>) {
     },
     // Use regular function (not arrow) because @fastify/multipart binds `this`
     // to the Fastify request in attachFieldsToBody mode.
-    async onFile(this: FastifyRequest, part) {
+    // Note: do NOT annotate `this` with the local Http2Server-typed FastifyRequest;
+    // the plugin types expect the default (RawServerDefault) FastifyRequest.
+    async onFile(part) {
       if (typeof this?.uploadDir !== 'string') {
         throw new Error('onFile: expected `this` to be bound to the Fastify request');
       }

